mne/decoding/search_light.py

```diff
@@ -0,0 +1,491 @@
+# Author: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from .mixin import TransformerMixin
+from .base import BaseEstimator  # XXX why are these sklearn object copied in weird places. We should just have an externals.sklearn? # noqa
+from ..parallel import parallel_func
+
+
+class SearchLight(BaseEstimator, TransformerMixin):
+    """Search Light.
+
+    Fit, predict and score a series of models to each subset of the dataset
+    along the third dimension.
+
+    Parameters
+    ----------
+    base_estimator : object | None (default=None)
+        The base estimator to iteratively fit on a subset of the dataset. If
+        None, then the estimator is a Logistic Regression.
```

Comment:
After some additional thoughts, I actually disagree. 

My original motivation was to make a generic object, not a redundancy of the GAT. Here are a few argument in favour of having no default estimator:

* Search lights can be used in many different contexts (across sensors, sources, freqs) to have a quick-and-dirty idea of where the information comes from. In some cases z-scoring the data isn't a great idea e.g.
    - when you want to keep angular values between 0 and 2 pi
    - when used after an Xdawn, it seems to be not such a great idea to rescale the components. 
    - when a scaling has already been applied per sensor (but not time point, e.g. after any spatial filtering), an additional z-score is an unnecessary step that increases memory load.

* in my analysis I very regularly do regression instead of classification. I don't see any good reason why the default of a search light should be either one or the other.

* search lights are analogous to ensemble or pipeline objects in sklearn, in the sense that they are meta-estimators. If we follow sklearn general design, such meta estimators either have no defaults (e.g. pipeline) or do, but are specifically named for sub cases (e.g.`BaggingClassifier` and `BaggingRegressor`).

* IMO a good lib is not just to provide tools that minimizes the number of lines users need to write, but also teaches how to design the analysis. In this view, I find it ok to make additional imports, and parameters specifications, for the same reason that I think it's fine to force users to specify `frequencies` in the single_trial_power decomposition.

* No default means less dev support, and simpler docs. (most users I interacted with had not realized that the default estimator was a pipelined scaler and logistic).

I'm obviously opened to discussion. Perhaps @dengemann, @jasmainak, @alexandrebarachant have some different experience/expectations?